service/src/test/java/uk/nhs/adaptors/gp2gp/ehr/mapper/EhrExtractMapperTest.java

```diff
@@ -80,11 +85,20 @@ public static void initialize() throws IOException {
             .build();
     }
 
+    @AfterAll
+    public static void deinitialize() {
+        TimeZone.setDefault(null);
+    }
+
     @BeforeEach
     public void setUp() {
         when(randomIdGeneratorService.createNewId()).thenReturn(TEST_ID);
```

Comment:
the test could return the list of values to verify if there aren't same records on the output.

Reply:
or, you can pre-populate the messageContext with correct id - it seems to be easier to maintain and verify your output xml is valid. Also if the input message contained less Encounters references, the output xml would be easier to test and maintain in the future. I also think it would be valuable to add test for empty list and when the list is not in the bundle.